spec/index.bs

```diff
@@ -5039,16 +5061,42 @@ offset {{GPUOrigin3D}} in texels, used when copying data from or to a {{GPUTextu
 
 Issue(gpuweb/gpuweb#69): Define the copies with {{GPUTextureDimension/1d}} and {{GPUTextureDimension/3d}} textures.
 
-### <dfn dictionary>GPUImageCopyImageBitmap</dfn> ### {#gpu-image-copy-image-bitmap-copy}
+### <dfn dictionary>GPUImageCopyPlatformImage</dfn> ### {#gpu-image-copy-source}
 
 <script type=idl>
-dictionary GPUImageCopyImageBitmap {
-    required ImageBitmap imageBitmap;
+dictionary GPUImageCopyPlatformImage {
+    required (ImageBitmap or HTMLVideoElement or HTMLCanvasElement or OffscreenCanvas) source;
     GPUOrigin2D origin = {};
 };
 </script>
 
-  * {{GPUImageCopyImageBitmap/origin}}: If unspecified, defaults to `[0, 0]`.
+{{GPUImageCopyPlatformImage}} has the following members:
+
+<dl dfn-type=dict-member dfn-for=GPUImageCopyPlatformImage>
+    : <dfn>source</dfn>
+    ::
+        The source of the image copy. The copy source data is captured at the moment that
+        {{GPUQueue/copyToTexture()}} is issued.
+
+    : <dfn>origin</dfn>
+    ::
+        Defines the origin of the copy - the minimum corner of the source sub-region to copy from.
```

Comment:
For HTMLVideoElment, it's a bit trick that it contains two types of ["width" and "height"](https://developer.mozilla.org/en-US/docs/Web/API/HTMLVideoElement).
- `HTMLVideoElment. width` and `HTMLVideoElement.height` is the display area
- `HTMLVideoElement.videoWidth` and `HTMLVideoElement.videoHeight` is the intrinsic width and height of the resource.

Myself support that when we use this API, we are talking about the intrinsic width and height of the resouce but need double check.

Reply:
There are also a bunch of different rectangles (coded size, natural size, etc), crops, rotations, etc. So we need to make sure to get this right and test it well. This behavior should most likely match WebGL exactly.